message-bus-tester/src/rabbit-message-queue/index.ts

```diff
@@ -0,0 +1,133 @@
+import { EventIdentifier, MessageQueue, Event } from "../message-queue";
+import * as amqplib from "amqplib";
+import { Connection, Message } from "amqplib";
+import { Option, Some } from "funfix";
+import { InfraLogger as logger } from "../logger";
+
+export class RabbitMessageQueue implements MessageQueue {
```

Comment:
Would it be good to split the queue implementation into `Publisher` and `Listener` ?

Reply:
:thinking:

Reply:
I don't know if it makes sense to split the implementation but I think it makes sense to split the interfaces into `Publisher` and `Subscriber` and maybe `EventBus`